gdb_backend/program.c

```diff
@@ -1,6 +1,270 @@
+	append(list, 19);
+	remove_back(list);
+
+
+	listd_t *dlist = new_listd(19.5);
+	appendd(dlist, 0.5);
+	appendd(dlist, 12);
+	remove_backd(dlist);
+	appendd(dlist, 12.765);
+	appendd(dlist, 43.095);
+	appendd(dlist, 23);
+	appendd(dlist, 12);
+	appendd(dlist, 23);
+	remove_val_d(dlist, 23);
+	remove_val_d(dlist, 12);
+	remove_val_d(dlist, 12);
+
+    struct list_2 *list2 = malloc(sizeof(struct list_2));
+    list2->head = NULL;
+    appendList(5, list2);
+    appendList(10, list2);
+    appendList(21, list2);
+    appendList(234, list2);
+    appendList(67, list2);
+    print_list(list2);
+}
+
+void func1(int *arr, int len, int modify) {
+	for (int i = 0; i < len; i++) {
+		if (i % 2 == 0) {
+			arr[i] += modify;
+		} else {
+			arr[i] *= modify;
+		}
+	}
+}
+
+
+double slow_pow(double base, int exponent) {
+	int res = 1;
+	for (int i = 0; i < exponent; i++) {
+		res *= base;
+	}
+
+	return res;
+}
+
+double fast_pow(double base, int exponent) {
+	if (exponent == 0) {
+		return 1;
+	} else if (exponent == 1) {
+		return base;
+	} else {
+		if (exponent % 2 == 0) {
+			double temp = fast_pow(base, exponent/2);
+			return temp * temp;
+		} else {
+			double temp = fast_pow(base, exponent/2);
+			return temp * temp * base;
+		}
+	}
+}
+
+// Basic list functions
+#pragma region basic list functions
+
+struct node *new_node(int val) {
+	struct node *new = malloc(sizeof(struct node));
+	new->data = val;
+	new->next = NULL;
+	return new;
+}
+
+struct node *new_list(int val) {
+	return new_node(val);
+}
+
+void append(struct node * list, int val) {
+	struct node *new = new_node(val);
+	struct node *curr = list;
+	while (curr->next != NULL) {
+		curr = curr->next;
+	}
+	curr->next = new;
+}
+
+void remove_back(struct node * list) {
+	struct node *curr = list;
+	while (curr->next->next != NULL) {
+		curr = curr->next;
+	}
+	free(curr->next);
+	curr->next = NULL;
+}
+
+void remove_val(struct node * list, int val) {
+	struct node *curr = list;
+	while (curr->next != NULL) {
+		if (curr->next->data == val) {
+			struct node *temp = curr->next->next;
+			free(curr->next);
+			curr->next = temp;
+			break;
+		}
+		curr = curr->next;
+	}
+}
+
+struct node *remove_head(struct node * list) {
+	struct node *temp = list->next;
+	free(list);
+	return temp;
+}
+#pragma endregion
+
+#pragma region contained list functions
+
+noded_t *new_noded(double val) {
+	noded_t *new = malloc(sizeof(noded_t));
+	new->data = val;
+	new->next = NULL;
+	new->prev = NULL;
+	return new;
+}
+
+listd_t *new_listd(double val) {
+	noded_t *elem = new_noded(val);
+	listd_t *new = malloc(sizeof(listd_t));
+	new->head = elem;
+	new->tail = elem;
+	new->size = 0;
+	return new;
+}
+
+void appendd(listd_t *list, double val) {
+	noded_t *elem = new_noded(val);
+	elem->prev = list->tail;
+	list->tail->next = elem;
+	list->tail = elem;
+	list->size++;
+}
+
+void remove_backd(listd_t *list) {
+	noded_t *temp = list->tail;
+	list->tail = list->tail->prev;
+	list->tail->next = NULL;
+	free(temp);
+}
```

Comment:
remember to update `list->size`

Reply:
need to account for edge cases in `void remove_val_d(listd_t *list, double val)`:
- list is initially empty
- remove first node
- remove last node